include/agent.h

```diff
@@ -0,0 +1,73 @@
+#ifndef AGENT_CPP_H
+#define AGENT_CPP_H
+
+#include <vector>
+
+#include "directive.h"
+
+
+namespace Rep
+{
+
+    class Agent
+    {
+    public:
+        /* The type for the delay. */
+        typedef float delay_t;
```

Comment:
Is there a particular reason you are creating this typedef?  It seems like you are adding an unnecessary level of obfuscation.